### PR TITLE
DOCS Remove archive download references

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-framework/issues/9232.

Also simplifies composer instructions a bit:
- Removes composer update --no-dev references, that's a bit of an edge case that people can just discover on getcomposer.org if they need it
- Changed example from the unused and oudated silverstripe/forum to silverstripe/blog
- Updated example versions to 4.x
- Remove "updating composer" section, it now tells you if its out of date
- Remove ss-auto-git-ignore module reference. The module hasn't been updated in ages, and it's much less necessary now that all relevant modules are on composer
- Add .env example config to getting started docs, I didn't realise it was stripped from the default --prefer-dist composer install

### CODE REVIEW

File: docs/en/05_Contributing/05_Making_A_SilverStripe_Core_Release.md [hunk 1]
```diff
@@ -346,47 +346,23 @@ Once this has been done, then the release is ready to be published live.
...
-* `--aws-profile <profile>` to specify the AWS profile name for uploading releases to s3. Check with
-  damian@silverstripe.com if you don't have an AWS key setup.
-* `--skip-archive-upload` to disable both "archive" and "upload". This is useful if doing a private release and
-  you don't want to upload this file to AWS.
-* `--skip-upload` to disable the "upload" command (but not archive)
-
-As with the `cow release` command, this step is broken down into the following
-subtasks which are invoked in sequence:
-
-* `release:tag` Each module will have the appropriate tag applied (except the theme). All tags are pushed up to origin
```
Comment (unclecheese) [inline docs/en/05_Contributing/05_Making_A_SilverStripe_Core_Release.md]: We still need this!

Comment (chillu) [reply docs/en/05_Contributing/05_Making_A_SilverStripe_Core_Release.md]: I figured that there's no point breaking down the command in the readme if there's only one subcommand now (`cow release:publish` runs `cow release:tag`, and nothing else).

**Comment by unclecheese** [comment]
I'd prefer to wait on https://github.com/silverstripe/silverstripe-framework/pull/9314 and rebase before moving this in, as we're right in the middle of a launch right now.

**Comment by chillu** [comment]
https://github.com/silverstripe/silverstripe-framework/pull/9314 is merged, does this mean this merge is good to go @unclecheese ?

**Comment by unclecheese** [comment]
Yup, a few conflicts to resolve, but all good now.

**Comment by unclecheese** [comment]
Spoke too soon.

> Using SilverStripe with Composer

Using ~~SilverStripe~~ Silverstripe CMS with Composer

Separate PR :)

**Pull request merged**